endpoints/data_objects/src/main.cpp

```diff
@@ -1790,17 +1805,84 @@ namespace
 						}
 						else if (iter->second == "update_existing") {
 							opts = fs::copy_options::update_existing;
+							addKeyVal(&input.destDataObjInp.condInput, FORCE_FLAG_KW, "");
 						}
 						else if (iter->second != "none") {
+							log::error("{}: Invalid value for [option] parameter.");
 							return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
 						}
 					}
 
+					const fs::path from = src_lpath_iter->second;
+					const fs::path to = dst_lpath_iter->second;
+
+					fs::throw_if_path_length_exceeds_limit(from);
+					fs::throw_if_path_length_exceeds_limit(to);
+
 					auto conn = irods::get_connection(client_info.username);
-					const auto copied =
-						fs::client::copy_data_object(conn, src_lpath_iter->second, dst_lpath_iter->second, opts);
 
-					res.body() = json{{"irods_response", {{"status_code", 0}}}, {"copied", copied}}.dump();
+					if (!fs::client::is_data_object(conn, from)) {
+						throw fs::filesystem_error{
+							"path does not point to a data object",
+							from,
+							irods::experimental::make_error_code(INVALID_OBJECT_TYPE)};
+					}
```

Comment:
> This is handled in the copy API if you wanted to avoid this check. Maybe this provides a better error message.

You're saying the API doesn't allow collections to be passed, right? Can you post the link showing that? I've certainly wondered whether some of the logic in the filesystem library could be removed.

And yes, it is also about consistent better/error messages.

> Also, consider adding the path in question to the error message.

Including the path in the error message deviates from the filesystem library, so can't change that here. Keep in mind this is a temporary solution to a bug in the 4.3.1 filesystem library.

However, we can change the log message in the catch clause to include the path.

Reply:
Sure. I was also going to open an issue that captures this change so we never forget to revisit.

Reply:
Done.